pkg/kubectl/stop.go

```diff
@@ -78,19 +78,26 @@ func (reaper *ReplicationControllerReaper) Stop(namespace, name string) (string,
 	return fmt.Sprintf("%s stopped", name), nil
 }
 
-func (reaper *PodReaper) Stop(namespace, name string) (string, error) {
+func (reaper *PodReaper) Stop(namespace, name string, gracePeriodSec int) (string, error) {
 	pods := reaper.Pods(namespace)
 	_, err := pods.Get(name)
 	if err != nil {
 		return "", err
 	}
-	if err := pods.Delete(name); err != nil {
-		return "", err
+	if gracePeriodSec >= 0 {
```

Comment:
DeleteWithOptions seems reasonable to me. We should just plumb that into all kinds.